build-support/bin/readme_api.py

```diff
@@ -0,0 +1,264 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
```

Comment:
`bin/` is pretty big and now will have 3 files who act as helpers/libraries for the actual script. Fine for not this change, but thoughts on `bin/utils/readme_api.py`? Better namespacing - `bin/` is only runnable scripts.

Reply:
Yeah, probably... Or maybe `build-support/bin` shouldn't exist at all, and we fold these under `src/python/pants`? It's weird to have this one separate "wild west" source root with no namespacing.

Reply:
Well, I do like `build-support` because it makes more clear that Pants's dist should never transitively depend on these scripts. Remember when we accidentally pulled in a dep on this generate_docs.py script and it resulted in that requests version conflict hell for plugins? 

But I think maybe the `/bin` part isn't necessary anymore and we could restructure the folder to be `build-support/generate_docs.py` which imports from `build-support/utils`, etc.

Reply:
I'd prefer that we enforced dependency restrictions via tooling, not via naming confusion... :)

Reply:
Isn't there a way to ban files to be pulled in, even if stuff depends on them. Like `!!pants.buildsupport` or something like that...